errors.go

```diff
@@ -2,28 +2,55 @@ package katzenmint
 
 import (
 	"fmt"
+
+	abcitypes "github.com/tendermint/tendermint/abci/types"
 )
 
 type KatzenmintError struct {
 	error
 
-	Msg  string
 	Code uint32
+	Msg  string
 }
 
 func (err KatzenmintError) Error() string {
 	return fmt.Sprintf("error (%d): %s", err.Code, err.Msg)
 }
 
 var (
-	ErrTxIsNotValidJSON     = KatzenmintError{Msg: "transaction is not valid json string", Code: 0x01}
-	ErrTxWrongPublicKeySize = KatzenmintError{Msg: "wrong public key size in transaction", Code: 0x02}
-	ErrTxWrongSignatureSize = KatzenmintError{Msg: "wrong public key size in transaction", Code: 0x03}
-	ErrTxWrongSignature     = KatzenmintError{Msg: "wrong signature in transaction", Code: 0x04}
+	// Transaction Common Errors
+	ErrTxIsNotValidJSON     = KatzenmintError{Code: 0x01, Msg: "transaction is not valid json string"}
+	ErrTxWrongPublicKeySize = KatzenmintError{Code: 0x02, Msg: "wrong public key size in transaction"}
+	ErrTxWrongSignatureSize = KatzenmintError{Code: 0x03, Msg: "wrong public key size in transaction"}
+	ErrTxWrongSignature     = KatzenmintError{Code: 0x04, Msg: "wrong signature in transaction"}
 
-	ErrTxDescInvalidVerifier   = KatzenmintError{Msg: "cannot get descriptor verifier", Code: 0x11}
-	ErrTxDescFalseVerification = KatzenmintError{Msg: "cannot verify and parse descriptor", Code: 0x12}
+	// Transaction Specific Errors
+	ErrTxDescInvalidVerifier    = KatzenmintError{Code: 0x11, Msg: "cannot get descriptor verifier"}
+	ErrTxDescFalseVerification  = KatzenmintError{Code: 0x12, Msg: "cannot verify and parse descriptor"}
+	ErrTxDescNotAuthorized      = KatzenmintError{Code: 0x13, Msg: "authority is not authorized"}
+	ErrTxDocFalseVerification   = KatzenmintError{Code: 0x14, Msg: "cannot verify and parse document"}
+	ErrTxDocEpoch               = KatzenmintError{Code: 0x15, Msg: "document epoch inconsistent with transaction epoch"}
+	ErrTxDocNotAuthorized       = KatzenmintError{Code: 0x16, Msg: "document is not authorized"}
+	ErrTxAuthorityParse         = KatzenmintError{Code: 0x17, Msg: "cannot parse authority"}
+	ErrTxAuthorityNotAuthorized = KatzenmintError{Code: 0x18, Msg: "descriptor is not authorized"}
+	ErrTxCommandNotFound        = KatzenmintError{Code: 0x19, Msg: "transaction command not found"}
 
-	ErrTxNonAuthorized   = KatzenmintError{Msg: "non authorized authority", Code: 0x31}
-	ErrTxCommandNotFound = KatzenmintError{Msg: "transaction command not found", Code: 0x32}
+	// Transaction Execution Errors
+	ErrTxExeEpoch   = KatzenmintError{Code: 0x21, Msg: "expect transaction epoch within +-1 to current epoch"}
```

Comment:
Maybe rename to `ErrWrongEpoch` or others.

Reply:
Fixed